tests/Aspire.Hosting.SqlServer.Tests/SqlServerFunctionalTests.cs

```diff
@@ -0,0 +1,284 @@
+
+using Aspire.Components.Common.Tests;
+using Aspire.Hosting.Utils;
+using Aspire.Hosting.Testing;
+using Microsoft.Data.SqlClient;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+using Polly;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Aspire.Hosting.SqlServer.Tests;
+
+public class SqlServerFunctionalTests(ITestOutputHelper testOutputHelper)
+{
+    private const string TestDbName = "testdb";
+    private const string MasterDbName = "master";
+
+    [Fact]
+    [RequiresDocker]
+    public async Task VerifySqlServerResource()
+    {
+        var cts = new CancellationTokenSource(TimeSpan.FromMinutes(5));
+        var pipeline = new ResiliencePipelineBuilder()
+            .AddRetry(new() { MaxRetryAttempts = 10, Delay = TimeSpan.FromSeconds(10) })
+            .Build();
+
+        var builder = CreateDistributedApplicationBuilder();
+
+        var sqlserver = builder.AddSqlServer("sqlserver");
+        var testDb = sqlserver.AddDatabase(TestDbName);
+        var masterDb = sqlserver.AddDatabase(MasterDbName);
+        using var app = builder.Build();
+
+        await app.StartAsync();
+
+        var hb = Host.CreateApplicationBuilder();
+
+        hb.Configuration.AddInMemoryCollection(new Dictionary<string, string?>
+        {
+            [$"ConnectionStrings:{testDb.Resource.Name}"] = await testDb.Resource.ConnectionStringExpression.GetValueAsync(default),
+            [$"ConnectionStrings:{masterDb.Resource.Name}"] = await masterDb.Resource.ConnectionStringExpression.GetValueAsync(default)
+        });
+
+        hb.AddKeyedSqlServerClient(masterDb.Resource.Name);
+        hb.AddSqlServerClient(testDb.Resource.Name);
+
+        using var host = hb.Build();
+
+        await host.StartAsync();
+
+        await pipeline.ExecuteAsync(async token =>
+        {
+            var masterDbConnection = host.Services.GetRequiredKeyedService<SqlConnection>(masterDb.Resource.Name);
+            var testDbConnection = host.Services.GetRequiredService<SqlConnection>();
+
+            await CreateTestDb(masterDbConnection, testDbConnection, token);
+
+        }, cts.Token);
+    }
+
+    [Theory]
+    [InlineData(true)]
+    [InlineData(false)]
+    [SkipOnCI("https://github.com/dotnet/aspire/issues/5055")]
+    [RequiresDocker]
+    public async Task WithDataShouldPersistStateBetweenUsages(bool useVolume)
+    {
+
+        string? volumeName = null;
+        string? bindMountPath = null;
+
+        var cts = new CancellationTokenSource(TimeSpan.FromMinutes(10));
+        var pipeline = new ResiliencePipelineBuilder()
+            .AddRetry(new() { MaxRetryAttempts = 10, Delay = TimeSpan.FromSeconds(10) })
+            .Build();
+
+        try
+        {
+            var builder1 = CreateDistributedApplicationBuilder();
+
+            var password = "P@ssw0rd1";
```

Comment:
Don't set passwords like this or this will be blocked on our internal CI. Check the other db functional tests, there is a way to read the default generated value.